package.json

```diff
@@ -18,13 +17,12 @@
     "should": "^1.3.0"
   },
   "scripts": {
-    "build": "npm-run-all build:esbuild build:typecheck build:copy-typings",
+    "build": "npm-run-all build:esbuild build:typecheck",
     "watch": "npm run build:esbuild -- --watch",
-    "test": "npm-run-all build test:unit-test",
-    "build:format": "prettier --write \"src/**/*.js\"",
+    "test": "npm-run-all build:format build test:unit-test",
+    "build:format": "prettier --write \"src/**/*.ts\"",
     "build:esbuild": "esbuild ./src/index.js --bundle --sourcemap --outdir=dist --platform=node",
-    "build:typecheck": "tsc --noEmit",
-    "build:copy-typings": "cpy ./index.d.ts ../dist/ --cwd=src",
+    "build:typecheck": "tsc --emitDeclarationOnly",
```

Comment:
TypeScript now generates the `.d.ts` files.

As a result, we don't need the manually maintained one any more, or the build logic to copy it over.

The downside of doing it this way is that TypeScript generates conrresponding `.d.ts` files for every source `.ts` file. I couldn't figure out config to include only the top level `index.ts` exports.

On the upside - its all automatically generated so 🤷 In the future you could not keep them in the repo and only generate them when publishing?

Reply:
Before deleting the old manually created one, I merged all the comments into the respective files.